plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridTableForeignKey.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.cubrid.model;
+
+import org.jkiss.dbeaver.ext.generic.model.GenericTableForeignKey;
+import org.jkiss.dbeaver.model.struct.DBSEntityReferrer;
+import org.jkiss.dbeaver.model.struct.rdb.DBSForeignKeyDeferability;
+import org.jkiss.dbeaver.model.struct.rdb.DBSForeignKeyModifyRule;
+
+public class CubridTableForeignKey extends GenericTableForeignKey {
+
+	public CubridTableForeignKey(CubridTable table, String name, String remarks, DBSEntityReferrer referencedKey,
+		DBSForeignKeyModifyRule deleteRule, DBSForeignKeyModifyRule updateRule,
+		DBSForeignKeyDeferability deferability, boolean persisted) {
+		super(table, name, remarks, referencedKey, deleteRule, updateRule, deferability, persisted);
+	}

```

Comment:
Is CubridTableForeignKey needed, it looks like we can just use GenericTableForeignKey

Reply:
Yes, It needed. If I use `GenericTableForeignKey`, `CubridForeignKeyManager` will be overridden by the `GenericForeignKeyManager` in configuration `plugin.xml` file.

Imagine if I use GenericTableForeignKey, the extension of Manager will be:

    <manager class="org.jkiss.dbeaver.ext.cubrid.edit.CubridForeignKeyManager" objectType="org.jkiss.dbeaver.ext.generic.model.GenericTableForeignKey"/>

It will override by generic plugin.xml when build.

    <manager class="org.jkiss.dbeaver.ext.generic.edit.GenericForeignKeyManager"    objectType="org.jkiss.dbeaver.ext.generic.model.GenericTableForeignKey"/>

Please let me know, if you have any solutions for this problem.

Reply:
We can try to use GenericKeyManager. It looks like there is no significant difference in constructors

Reply:
I have fixed it by removing CubridTableForeignKey and using GenericTableForeignKey instead.